tests/compat/test_gecko50.py

```diff
@@ -0,0 +1,139 @@
+import pytest
+
+from helper import CompatTestCase
+from validator.compat import FX50_DEFINITION
+
+
+class TestFX50Compat(CompatTestCase):
+    """Test that compatibility tests for Gecko 50 are properly executed."""
+
+    VERSION = FX50_DEFINITION
+
+    def test_nsIX509Cert_removed(self):
+        """https://github.com/mozilla/addons-server/issues/3408"""
+        expected = (
+            'The methods getUsagesArray, requestUsagesArrayAsync, and '
+            'getUsagesString are no longer supported.')
+
+        script = '''
+            var certdb = Components.classes["@mozilla.org/security/x509certdb;1"].
+                getService(Components.interfaces.nsIX509CertDB);
+            var cert = certdb.findCertByDBKey(
+                tgt.parentNode.firstChild.getAttribute("display"),
+                null);
+
+            cert.QueryInterface(Components.interfaces.nsIX509Cert);
+            cert.getUsagesArray();
+        '''
+
+        self.run_script_for_compat(script)
+
+        assert not self.compat_err.notices
+        assert not self.compat_err.errors
+
+        error = [msg for msg in self.compat_err.warnings if msg['message'] == expected]
+        assert error
```

Comment:
Wrong name, indeed but it actually contains a warning. I'll fix it


On Wed, Sep 7, 2016, at 11:50 AM, Andreas Wagner wrote:
> In tests/compat/test_gecko50.py[1]:


> > +                getService(Components.interfaces.nsIX509CertDB); >
> >                  +            var cert = certdb.findCertByDBKey( > +
> >                  tgt.parentNode.firstChild.getAttribute("display"),
> >                  > +                null); > + > +            cert.-
> >                  QueryInterface(Components.interfaces.nsIX509Cert);
> >                  > +            cert.getUsagesArray(); > +
> >                  ''' > + > +
> >                  self.run_script_for_compat(script) > + > +
> >                  assert not self.compat_err.notices > +
> >                  assert not self.compat_err.errors > + > +
> >                  error = [msg for msg in self.compat_err.warnings if
> >                  msg['message'] == expected] > +        assert error
>
> This should be a warning, no?
> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[2], or mute the thread[3].
>

Links:

  1. https://github.com/mozilla/amo-validator/pull/466#discussion_r77792196
  2. https://github.com/mozilla/amo-validator/pull/466/files/9a22865ee6c3ba4a2fff61fa57e358f8005e4cab#r77792196
  3. https://github.com/notifications/unsubscribe-auth/AAIfGfQy3c6j38tXt4OqJ0bZmr15sgsjks5qnojogaJpZM4J2s5A